src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java

```diff
@@ -278,26 +258,48 @@ private ClientTestFileView createUnitTestFileView(SurfaceTransformerContext cont
     return testFile.build();
   }
 
-  private List<ClientTestCaseView> createTestCaseViews(SurfaceTransformerContext context) {
-    ArrayList<ClientTestCaseView> testCaseViews = new ArrayList<>();
+  private List<TestMethodView> createTestCaseViews(SurfaceTransformerContext context) {
+    ArrayList<TestMethodView> testCaseViews = new ArrayList<>();
     SymbolTable testNameTable = new SymbolTable();
     for (Method method : context.getSupportedMethods()) {
       MethodConfig methodConfig = context.getMethodConfig(method);
-      if (MethodConfig.isGrpcStreamingMethod(method)) {
+      if (methodConfig.isGrpcStreaming()) {
         addGrpcStreamingTestImport(context);
         MethodTransformerContext methodContext = context.asRequestMethodContext(method);
         testCaseViews.add(
-            createTestCaseView(
-                methodContext, testNameTable, methodConfig.getRequiredFieldConfigs()));
+            mockServiceTransformer.createTestMethodView(
+                methodContext,
+                testNameTable,
+                initCodeTransformer.createRequestInitCodeContext(
+                    methodContext,
+                    new SymbolTable(),
+                    methodConfig.getRequiredFieldConfigs(),
+                    InitCodeOutputType.SingleObject,
+                    valueGenerator),
+                ClientMethodType.CallableMethod));
       } else if (methodConfig.isFlattening()) {
+        ClientMethodType clientMethodType;
```

Comment:
Maybe worth moving this logic into methodConfig so we can just call methodConfig.getMethodType() here.